code/game/objects/items/tools/weldingtool.dm

```diff
@@ -151,6 +141,24 @@
 
 	update_icon()
 
+/obj/item/weldingtool/proc/handle_mob_ignite(atom/O, mob/user, proximity)
+	if(!proximity)
+		return
+	
+	if(isOn())
+		handle_fuel_and_temps(1, user)
+		
+		if(!QDELETED(O) && isliving(O)) // can't ignite something that doesn't exist
+			var/mob/living/L = O
+			if(L.IgniteMob())
+				message_admins("[ADMIN_LOOKUPFLW(user)] set [key_name_admin(L)] on fire with [src] at [AREACOORD(user)]")
+				log_game("[key_name(user)] set [key_name(L)] on fire with [src] at [AREACOORD(user)]")
+
+// Ah fuck, I can't believe you've done this
+/obj/item/weldingtool/proc/handle_fuel_and_temps(used = 0, mob/living/user)
```

Comment:
why dis a proc?

Reply:
Duplicate code

Reply:
If you'd rather me copy paste it into each th ats fine

Reply:
you can try to define it or just copypasta, procs are not free.

Reply:
guess I'll copy paste, there is no way I'm trying to define that

Reply:
there you go